### PR TITLE
Upload iOS Framework

### PR DESCRIPTION
**Pull request by maxxfrazer**
I've added a new workflow file that builds the frameworks, uploads to the build process, and adds them to a release when a tag is added.

With no readme it was a bit tricky, as I've not used rust before. Please let me know if anything doesn't look right in the workflow.

I'm adding this because I also want to add version control of this framework on the swift package, rather than uploading the framework directly to the repo.

### CODE REVIEW

**Comment by maxxfrazer** [comment]
@NanYeZhuYu @renfengshi @LiYanLance @zhaomengru2015 please let me know who's best to take a look at this! cheers

**Comment by LiYanLance** [comment]
Hey @maxxfrazer , thanks for the pull request.

> I'm adding this because I also want to add version control of this framework on the swift package, rather than uploading the framework directly to the repo.

This is great and exactly what we wanna do.

We had a look and think there are some contexts you might want to know
1. The Android SDK is using the rust code in another way, this repo is used via the git submodule in Android SDK, and rust code gets built in the process of building the Android library, check the [build script](https://github.com/KeystoneHQ/keystone-sdk-android/blob/master/library/build.sh). Can you remove the change of the Android part in the PR until we have time to uplift the process of building the Android SDK? We will add a separate workflow for the Android release if needed later.
2. Currently, the repo is not only used by mobile SDKs but our hardware wallet Keystone. So the tags we have now are actually for UR, which makes the release rule of Android and iOS in your code doesn't make much sense. Could you add a tag prefix check for releasing, e.g. starts with `sdk`?

**Comment by maxxfrazer** [comment]
Thanks @LiYanLance! I've moved the Android parts, and updated the script to only run on tags starting with sdk. I'll run a quick check on my fork to see that it works right and will link to them.

**Comment by maxxfrazer** [comment]
Did a test with tags `sdk-0.0.1`, `sdk/iOS/0.0.1` and `different/ios/0.0.1`. As I was hoping, the job only runs for the first two:

Screenshot 2023-05-29 at 13 33 56
Screenshot 2023-05-29 at 13 33 40

https://github.com/maxxfrazer/keystone-sdk-rust/actions

File: .github/workflows/ios-build-release.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    tags:  
+      # Push events to every tag starting with "sdk", including those with /     
+      - 'sdk**'
+  pull_request:
+    branches:
+      - 'master'
+
+jobs:
+    build_ios:
+        name: Build Android+iOS Frameworks
```
Comment (LiYanLance) [inline .github/workflows/ios-build-release.yml:14]: Could you remove `Android` here as well?

**Review by LiYanLance — COMMENTED** [review]
Looks great.

One last thing, it seems the step `Upload XCFramework to Release` failed in the build you mentioned above, due to no tag supplied, could you take a look?

**Comment by maxxfrazer** [comment]
> One last thing, it seems the step `Upload XCFramework to Release` failed in the build you mentioned above, due to no tag supplied, could you take a look?

Weird, I was passing github.event.base_ref as the tag, but maybe it's null on tag releases.

trying just `github.ref` instead now:
https://github.com/maxxfrazer/keystone-sdk-rust/actions/runs/5109176613/jobs/9183703765

**Comment by maxxfrazer** [comment]
another test incoming, fire alarm test in my building so power's out right now 🥴

**Comment by maxxfrazer** [comment]
@LiYanLance workflow permissions were blocking it too, but all good now:
https://github.com/maxxfrazer/keystone-sdk-rust/actions/runs/5109420075/jobs/9184193939

**Comment by LiYanLance** [comment]
We are almost there except the checksum in the release body didn't get computed. 
https://github.com/maxxfrazer/keystone-sdk-rust/releases.

Haven't find a way to do calculation in `body` field, but I feel like you can use variable according to [their doc](https://github.com/svenstaro/upload-release-action/blob/master/README.md)

```yaml
body: |
    ${{ steps.read_release.outputs.RELEASE_BODY }}  # <--- Use environment variables that was created earlier
```

**Comment by maxxfrazer** [comment]
Drat, thanks for spotting that. Doing hopefully a final test now 🤞 getting late here (GMT+8), and not sure I'll be able to dedicate much time tomorrow, so I really hope it works!!

https://github.com/maxxfrazer/keystone-sdk-rust/actions/runs/5111968912

**Review by LiYanLance — APPROVED** [review]

**Comment by LiYanLance** [comment]
@maxxfrazer  It works, thanks for the improvement. 👍  Merged.

**Pull request merged**